sim/src/test/scala/midasexamples/TutorialSuite.scala

```diff
@@ -192,6 +192,45 @@ class ParityVitisTest extends TutorialSuite("Parity", platformConfigs = classOf[
   runTest("verilator", true)
   runTest("vcs", true)
 }
+
+
+/** Trait so that we have a uniform numbering scheme for the plusargs tests
+  */
+trait PlusargsKey {
+  def getKey(groupNumber: Int, testNumber: Int): String = {
+    val key = (groupNumber << 4 | testNumber)
+    s"+plusargs_test_key=${key}"
+  }
+}
+
+class PlusargsGroup0 extends TutorialSuite("PlusargsModule", "PlusargsModuleTestConfigGroup0") with PlusargsKey {
```

Comment:
On CI, each of these tests takes tens of minutes. Could they be grouped into a single verilator run somehow?

Reply:
I'll talk this over with @davidbiancolin and we can probably come up with something faster

Reply:
@nandor With the code as is, when I run `testOnly firesim.midasexamples.PlusargsGroup*` I get a total run time of 4 minutes. I'm using a `c5.4xlarge`.

I did a rough breakdown and this is what I found (all times are approx):
```
elaboration: 57 seconds
verilation:  25 seconds
test run:    5 seconds
```

The code you reviewed has 4 verilations. I discovered a simple change to reduce this down to 2 verilations.

I do not see where you are getting "Tens of minutes". This number is large and concerning. I think we should track down the issue here and agree on run times before we do further optimizations.

Reply:
This is fine - talked to @sifive-benjamin-morse offline.